### PR TITLE
Fix/refactor decode_base64_field processor (libbeat)

### PR DESCRIPTION
**Pull request by guyromb**
Fixes:
- Key is not fetched correctly
- Clean duplicated doc key in processors-using
- Aligning AllowedFields according to spec
- General cleaning/refactoring

#docs #review #needs_backport #libbeat #bug

### CODE REVIEW

**Comment by guyromb** [comment]
run jenkins

**Comment by guyromb** [comment]
jenkins, test this please

**Comment by guyromb** [comment]
Could anyone from elastic be kind to review && hotfix this? This feature is broken.
@dedemorton @ph

File: libbeat/processors/actions/decode_base64_field.go [hunk 1]
```diff
@@ -46,34 +45,28 @@ type base64Config struct {
...
-		IgnoreMissing: false,
-		FailOnError:   true,
-	}
-)
-
 func init() {
 	processors.RegisterPlugin(processorName,
 		checks.ConfigChecked(NewDecodeBase64Field,
 			checks.RequireFields("field"),
-			checks.AllowedFields("field", "when")))
```
Comment (kvch) [inline libbeat/processors/actions/decode_base64_field.go]: Please don't delete `when`. It is required for conditional processing.

File: libbeat/processors/actions/decode_base64_field.go [hunk 2]
```diff
@@ -84,55 +77,55 @@ func (f *decodeBase64Field) Run(event *beat.Event) (*beat.Event, error) {
...
 	if err != nil {
-		// Ignore ErrKeyNotFound errors
 		if f.config.IgnoreMissing && errors.Cause(err) == common.ErrKeyNotFound {
 			return nil
 		}
-		return fmt.Errorf("could not fetch value for key: %s, Error: %s", from, err)
+		return fmt.Errorf("could not fetch base64 value for key: %s, Error: %v", f.config.Field.From, err)
 	}
 
-	text, ok := value.(string)
```
Comment (kvch) [inline libbeat/processors/actions/decode_base64_field.go]: Please do not rename `ok`, as it's a Golang convention when doing type assertions.

File: libbeat/docs/processors-using.asciidoc [hunk 3]
```diff
@@ -215,7 +215,6 @@ ifdef::has_decode_csv_fields_processor[]
  * <<decode-csv-fields,`decode_csv_fields`>>
 endif::[]
  * <<decode-json-fields,`decode_json_fields`>>
- * <<decode-base64-field,`decode_base64_field`>>
```
Comment (kvch) [inline libbeat/docs/processors-using.asciidoc]: Why did you remove this reference?

File: libbeat/processors/actions/decode_base64_field.go [hunk 4]
```diff
@@ -84,55 +77,55 @@ func (f *decodeBase64Field) Run(event *beat.Event) (*beat.Event, error) {
...
+		return fmt.Errorf("error trying to decode %s: %v", base64String, err)
 	}
 
-	field := to
+	target := f.config.Field.To
 	// If to is empty
-	if to == "" || from == to {
+	if f.config.Field.To == "" || f.config.Field.From == f.config.Field.To {
 		// Deletion must happen first to support cases where a becomes a.b
-		if err = fields.Delete(from); err != nil {
```
Comment (kvch) [inline libbeat/processors/actions/decode_base64_field.go]: Do you mind removing this `if`? It's not required in this processor, but got copied from rename processor.

File: libbeat/docs/processors-using.asciidoc (hunk 3 above)
Comment (guyromb) [reply libbeat/docs/processors-using.asciidoc]: It is duplicated :-) I just removed the second reference.

File: libbeat/processors/actions/decode_base64_field.go (hunk 1 above)
Comment (guyromb) [reply libbeat/processors/actions/decode_base64_field.go]: fixed

File: libbeat/processors/actions/decode_base64_field.go (hunk 2 above)
Comment (guyromb) [reply libbeat/processors/actions/decode_base64_field.go]: fixed

File: libbeat/processors/actions/decode_base64_field.go (hunk 4 above)
Comment (guyromb) [reply libbeat/processors/actions/decode_base64_field.go]: done

**Comment by guyromb** [comment]
@kvch I addressed the issues. 
Do you know if to support base64 processor using hints (k8s auto discovery) we also need to do any additional changes?

**Comment by kvch** [comment]
jenkins test this

**Comment by kvch** [comment]
Tested locally, seems to work,

**Comment by kvch** [comment]
It seems that the error messages in the unit test need to be adjusted.

**Comment by guyromb** [comment]
On it @kvch

**Comment by kvch** [comment]
jenkins test this

**Comment by guyromb** [comment]
Seems like it fails due to unrelated things (connection timed out) both on travis and jenkins.

**Comment by kvch** [comment]
@guyromb Thank you for your contribution!
